flink-yarn/src/main/java/org/apache/flink/yarn/configuration/YarnConfigOptions.java

```diff
@@ -52,7 +52,74 @@
 		key("yarn.per-job-cluster.include-user-jar")
 			.defaultValue("ORDER");
 
+	/**
+	 * The vcores exposed by YARN.
+	 */
+	public static final ConfigOption<Integer> YARN_VCORES =
+		key("yarn.containers.vcores")
+		.defaultValue(Integer.MAX_VALUE);
+
+	/**
+	 * The maximum number of failed YARN containers before entirely stopping
+	 * the YARN session / job on YARN.
+	 *
+	 * By default, we take the number of of initially requested containers.
+	 */
+	public static final ConfigOption<String> YARN_MAX_FAILED_CONTAINERS =
+		key("yarn.maximum-failed-containers")
+		.noDefaultValue();
 
+	/**
+	 * Set the number of retries for failed YARN ApplicationMasters/JobManagers in high
+	 * availability mode. This value is usually limited by YARN.
+	 *
+	 * By default, it's 1 in the standalone case and 2 in the high availability case.
+	 */
+	public static final ConfigOption<String> YARN_APPLICATION_ATTEMPTS =
+		key("yarn.application-attempts")
+		.noDefaultValue();
+
+	/**
+	 * The heartbeat interval between the Application Master and the YARN Resource Manager.
+	 *
+	 * The default value is 5 (seconds).
+	 */
+	public static final ConfigOption<String> YARN_HEARTBEAT_DELAY_SECONDS =
+		key("yarn.heartbeat-delay")
+		.defaultValue("5 s");
```

Comment:
We can't just change the type of the option, nor the syntax with which it is configured. It should remain an Integer.